packages/compiler/core/parser.ts

```diff
@@ -866,6 +895,50 @@ function createParser(code: string | SourceFile, options: ParseOptions = {}): Pa
     };
   }
 
+  function parseModelPropertyOrValidate(
+    pos: number,
+    decorators: DecoratorExpressionNode[]
+  ): ModelPropertyNode | ModelValidateNode {
+    const id = parseIdentifier({
+      message: "property",
+      allowStringLiteral: true,
+    });
+
+    if (id.sv === "validate" && token() === Token.Identifier) {
+      const vid = parseIdentifier({
```

Comment:
I think we should make the `id` optional. It is super annoying to have to add an id, but will be needed depending on how you want to use it. Linter can of course enforce that you have `id`.